streams/src/main/java/org/apache/kafka/streams/kstream/internals/GroupedStreamAggregateBuilder.java

```diff
@@ -75,8 +75,18 @@ public V apply() {
                            final boolean isQueryable) {
         final String aggFunctionName = builder.newProcessorName(functionName);
         final String sourceName = repartitionIfRequired(storeBuilder.name());
-        builder.internalTopologyBuilder.addProcessor(aggFunctionName, aggregateSupplier, sourceName);
-        builder.internalTopologyBuilder.addStateStore(storeBuilder, aggFunctionName);
+
+        StatefulProcessorNode.StatefulProcessorNodeBuilder<K, V> statefulProcessorNodeBuilder = StatefulProcessorNode.statefulProcessorNodeBuilder();
+        ProcessorParameters processorParameters = new ProcessorParameters<>(aggregateSupplier, functionName);
+        statefulProcessorNodeBuilder
+            .withProcessorParameters(processorParameters)
+            .withNodeName(aggFunctionName)
+            .withRepartitionRequired(repartitionRequired)
+            .withStoreBuilder(storeBuilder)
+            .withMaybeRepartitionedSourceName(sourceName);
+
+        builder.addNode(statefulProcessorNodeBuilder.build());
```

Comment:
Why we are adding more nodes here? Shouldn't any stream graph nodes be added earlier??

Reply:
These nodes are child nodes of the previously created nodes, contain the information needed for adding the aggregation store and processor